### PR TITLE
HPCC-16440 Clear up some minor issues spotted by Coverity

### PR DESCRIPTION
**Pull request by jakesmith**
- False positive warning re. uninitialized 'type' var.
- Unreachable code warnings

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@AttilaVamos - please review

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -9162,10 +9162,10 @@ class CInitGroups
         if (!newClusterGroup) // both null
             return true;
         // see if identical
-        const char *oldKind = oldClusterGroup?oldClusterGroup->queryProp("@kind"):NULL;
```
Comment (AttilaVamos) [inline dali/base/dadfs.cpp]: The checking between lines 9158 and 9163 seems a bit strange.

**Comment by AttilaVamos** [comment]
@jakesmith The changes seems fine. Only one comment unrelated to changed code.

File: dali/base/dadfs.cpp (hunk 1 above)
Comment (jakesmith) [inline dali/base/dadfs.cpp]: It is, it's basically saying if both null - then they match - if either null then mismatch.

Comment (AttilaVamos) [inline dali/base/dadfs.cpp]: OK. Thanks.

Comment (jakesmith) [inline dali/base/dadfs.cpp]: @AttilaVamos - I've modified to make clearer, please check.

**Comment by AttilaVamos** [comment]
@jakesmith thanks, looks good.

**Comment by jakesmith** [comment]
@richardkchapman - please merge.

**Pull request merged**